base/range.jl

```diff
@@ -143,15 +144,15 @@ function colon{T<:AbstractFloat}(start::T, step::T, stop::T)
     if lo <= step <= hi
         a0, b = rat(start)
         a = convert(T,a0)
-        if a/convert(T,b) == start
+        if a/convert(T,b) == unitless(start)
```

Comment:
I hate having to add the extra `unitless` function. Dividing by unity is still not pretty, but the compiler is generally able to optimize it away and it should have the same effect.